protostar/testing/environments/fuzz_test_execution_environment.py

```diff
@@ -50,6 +50,9 @@ def __init__(self, state: TestExecutionState):
         self.given_strategies: dict[str, SearchStrategy] = {}
 
     async def execute(self, function_name: str) -> FuzzTestExecutionResult:
+        assert (
+            self.state.contract
+        ), "Tried to use legacy execution flow with no test contract provided!"
```

Comment:
If TestExecutionState required contract instead of ExecutionState we could avoid this assert